hs_core/tasks.py

```diff
@@ -461,17 +464,15 @@ def create_new_version_resource_task(ori_res_id, username, new_res_id=None):
         utils.copy_and_create_metadata(ori_res, new_res)
 
         # add or update Relation element to link source and target resources
-        hs_identifier = new_res.metadata.identifiers.all().filter(name="hydroShareIdentifier")[0]
-        ori_res.metadata.create_element('relation', type='isReplacedBy', value=hs_identifier.url)
+        ori_res.metadata.create_element('relation', type='isReplacedBy', value=new_res.get_citation())
 
         if new_res.metadata.relations.all().filter(type='isVersionOf').exists():
             # the original resource is already a versioned resource, and its isVersionOf relation
             # element is copied over to this new version resource, needs to delete this element so
             # it can be created to link to its original resource correctly
             new_res.metadata.relations.all().filter(type='isVersionOf').first().delete()
 
-        hs_identifier = ori_res.metadata.identifiers.all().filter(name="hydroShareIdentifier")[0]
-        new_res.metadata.create_element('relation', type='isVersionOf', value=hs_identifier.url)
+        new_res.metadata.create_element('relation', type='isVersionOf', value=ori_res.get_citation())
```

Comment:
Same comment as above - why use the resource citation rather than resource identifier here?

Reply:
see my reply above.

Reply:
Got it. I kind of figured out this must be the requirement after reviewing more code changes.